src/main/java/org/medicmobile/webapp/mobile/MedicAndroidJavascript.java

```diff
@@ -230,6 +254,96 @@ public void sms_send(String id, String destination, String content) throws Excep
 		}
 	}
 
+	@org.xwalk.core.JavascriptInterface
+	@android.webkit.JavascriptInterface
+	public String getDeviceInfo() {
+		try {
+			if (activityManager == null) {
+				return jsonError("ActivityManager not set. Cannot retrieve RAM info.");
+			}
+
+			if (connectivityManager == null) {
+				return jsonError("ConnectivityManager not set. Cannot retrieve network info.");
+			}
+
+			String versionName = parent.getPackageManager()
+					.getPackageInfo(parent.getPackageName(), 0)
+					.versionName;
+			JSONObject appObject = new JSONObject();
+			appObject.put("version", versionName);
+
+			String androidVersion = Build.VERSION.RELEASE;
+			int osApiLevel = Build.VERSION.SDK_INT;
+			String osVersion = System.getProperty("os.version") + "(" + android.os.Build.VERSION.INCREMENTAL + ")";
+			JSONObject softwareObject = new JSONObject();
+			softwareObject
+					.put("androidVersion", androidVersion)
+					.put("osApiLevel", osApiLevel)
+					.put("osVersion", osVersion);
+
+			String device = Build.DEVICE;
+			String model = Build.MODEL;
+			String manufacturer = Build.BRAND;
+			String hardware = Build.HARDWARE;
+			Map<String, String> cpuInfo = getCPUInfo();
+			JSONObject hardwareObject = new JSONObject();
+			hardwareObject
+					.put("device", device)
+					.put("model", model)
+					.put("manufacturer", manufacturer)
+					.put("hardware", hardware)
+					.put("cpuInfo", new JSONObject(cpuInfo));
+
+			File dataDirectory = Environment.getDataDirectory();
+			StatFs dataDirectoryStat = new StatFs(dataDirectory.getPath());
+			long dataDirectoryBlockSize = dataDirectoryStat.getBlockSizeLong();
+			long dataDirectoryAvailableBlocks = dataDirectoryStat.getAvailableBlocksLong();
+			long dataDirectoryTotalBlocks = dataDirectoryStat.getBlockCountLong();
+			long freeMemorySize = dataDirectoryAvailableBlocks * dataDirectoryBlockSize;
+			long totalMemorySize = dataDirectoryTotalBlocks * dataDirectoryBlockSize;
+			JSONObject memoryObject = new JSONObject();
+			memoryObject
+					.put("free", freeMemorySize)
+					.put("total", totalMemorySize);
+
+			MemoryInfo memoryInfo = new ActivityManager.MemoryInfo();
+			activityManager.getMemoryInfo(memoryInfo);
+			long totalRAMSize = memoryInfo.totalMem;
+			long freeRAMSize = memoryInfo.availMem;
+			long thresholdRAM = memoryInfo.threshold;
+			Runtime runtime = Runtime.getRuntime();
+			long maxMemorySize = runtime.maxMemory();
+			JSONObject ramObject = new JSONObject();
+			ramObject
+					.put("free", freeRAMSize)
+					.put("total", totalRAMSize)
+					.put("threshold", thresholdRAM)
+					.put("maxMemory", maxMemorySize);
```

Comment:
If we decide to keep this, but only this from the runtime, it should be `jvmMaxMemory` or `runtimeMaxMemory` to avoid confusion with the rest of the system information.

If we decide to keep this and expand to the other information we should have a new section for `jvm` or `runtime` information.